native/src/main/java/io/ballerina/lib/ldap/Client.java

```diff
@@ -91,28 +99,28 @@ public static BError initLdapConnection(BObject ldapClient, BMap<BString, Object
         return null;
     }
 
-    public static Object add(Environment env, BObject ldapClient,
-                             BString distinguishedName, BMap<BString, Object> entry) {
+    public static Object add(Environment env, BObject ldapClient, BString dN, BMap<BString, Object> entry) {
         Future future = env.markAsync();
         try {
             LDAPConnection ldapConnection = (LDAPConnection) ldapClient.getNativeData(NATIVE_CLIENT);
             validateConnection(ldapConnection);
-            AddRequest addRequest = generateAddRequest(distinguishedName, entry);
+            AddRequest addRequest = generateAddRequest(dN, entry);
             CustomAsyncResultListener customAsyncResultListener = new CustomAsyncResultListener(future);
             ldapConnection.asyncAdd(addRequest, customAsyncResultListener);
+        } catch (LDAPException e) {
+            future.complete(Utils.createError(e.getMessage(), e));
         } catch (Exception e) {
             future.complete(Utils.createError(e.getMessage(), e));
         }
```

Comment:
Is there a reason why we can't merge these 2 catch blocks?

Reply:
Here we use overloaded methods for `Utils.createError()` with `LDAPException` and `Exception` types. Java doesn't allow merging them into one catch block, since `Exception` is the super class of `LDAPException`

Reply:
Got it. It looked like we are calling the same method inside both blocks :) +1

Reply:
> Here we use overloaded methods for `Utils.createError()` with `LDAPException` and `Exception` types. Java doesn't allow merging them into one catch block, since `Exception` is the super class of `LDAPException`

So the question is do we really need to catch `Exception` here? :)

Reply:
Removed unnecessary the catch blocks 0f3fc74